File: src/DataScience/R/facade.R
auth_service <- function(user) {
  sprintf("auth(%s)", user)
}

inventory_service <- function(sku) {
  sprintf("reserve(%s)", sku)
}

billing_service <- function(amount) {
  sprintf("charge(%d)", amount)
}

checkout_facade <- function(user, sku, amount) {
  steps <- c(
    auth_service(user),
    inventory_service(sku),
    billing_service(amount)
  )
  paste0("checkout=", paste(steps, collapse = ">"))
}

cat(checkout_facade("alice", "SKU-42", 499), "\n", sep = "")

File: src/DataScience/Julia/facade.jl
auth_service(user) = "auth($(user))"
inventory_service(sku) = "reserve($(sku))"
billing_service(amount) = "charge($(amount))"

function checkout_facade(user, sku, amount)
    steps = (
        auth_service(user),
        inventory_service(sku),
        billing_service(amount),
    )
    return "checkout=" * join(steps, ">")
end

println(checkout_facade("alice", "SKU-42", 499))
